deepchecks/checks/distribution/train_test_label_drift.py

```diff
@@ -0,0 +1,230 @@
+from deepchecks.utils.typing import Hashable
+import plotly.graph_objects as go
+from plotly.subplots import make_subplots
+
+__all__ = ['TrainTestLabelDrift']
+
+
+class TrainTestLabelDrift(TrainTestBaseCheck):
+    """
+    Calculate label drift between train dataset and test dataset, using statistical measures.
+
+    Check calculates a drift score for the label in test dataset, by comparing its distribution to the train
+    dataset.
+    For numerical columns, we use the Earth Movers Distance.
+    See https://www.lexjansen.com/wuss/2017/47_Final_Paper_PDF.pdf
+    For categorical columns, we use the Population Stability Index (PSI).
+    See https://en.wikipedia.org/wiki/Wasserstein_metric.
+
+
+    Args:
+        max_num_categories (int):
+            Only for categorical columns. Max number of allowed categories. If there are more,
+            they are binned into an "Other" category. If max_num_categories=None, there is no limit. This limit applies
+            for both drift calculation and for distribution plots.
+    """
+
+    def __init__(
+            self,
+            max_num_categories: int = 10
+    ):
+        super().__init__()
+        self.max_num_categories = max_num_categories
+
+    def run(self, train_dataset, test_dataset, model=None) -> CheckResult:
+        """Run check.
+
+        Args:
+            train_dataset (Dataset): The training dataset object.
+            test_dataset (Dataset): The test dataset object.
+            model: not used in this check.
+
+        Returns:
+            CheckResult:
+                value: dictionary of column name to drift score.
+                display: distribution graph for each column, comparing the train and test distributions.
+
+        Raises:
+            DeepchecksValueError: If the object is not a Dataset or DataFrame instance
+        """
+        return self._calc_drift(train_dataset, test_dataset)
+
+    def _calc_drift(
+            self,
+            train_dataset: Dataset,
+            test_dataset: Dataset,
+    ) -> CheckResult:
+        """
+        Calculate drift for all columns.
+
+        Args:
+            train_dataset (Dataset): The training dataset object. Must contain a label.
+            test_dataset (Dataset): The test dataset object. Must contain a label.
+
+        Returns:
+            CheckResult:
+                value: drift score.
+                display: label distribution graph, comparing the train and test distributions.
+        """
+        train_dataset = Dataset.validate_dataset(train_dataset)
+        test_dataset = Dataset.validate_dataset(test_dataset)
+
+        drift_score, method, display = self._calc_drift_per_column(
+            train_column=train_dataset.label_col,
+            test_column=test_dataset.label_col,
+            column_name=train_dataset.label_name,
+            column_type='categorical' if train_dataset.label_type == 'classification_label' else 'numerical',
```

Comment:
If a model was passed, we should use it to determine the type

Reply:
Disagree. This is purely on the data and the user can better control the type if inferred from dataset / directly given

Reply:
mm well can concede, if only on the grounds of this not running with a model most of the time